### PR TITLE
fix(BS-15125) restore call to isInvolvedInProcessInstance Method

### PR DESCRIPTION
**Pull request by abirembaut**
- restore call to isInvolvedInProcessInstance engine method in 6.x forms
for process instance overview access control

Covers [BS-15125](https://bonitasoft.atlassian.net/browse/BS-15125)

### CODE REVIEW

**Comment by abirembaut** [comment]
waiting for engine fix

File: forms/forms-server/src/main/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImpl.java [hunk 1]
```diff
@@ -900,10 +900,9 @@ public Boolean canUserSeeHumanTask(final APISession session, final long userId,
     @Override
     public boolean canUserSeeProcessInstance(final APISession session, final long processInstanceID)
             throws ProcessInstanceNotFoundException, BPMEngineException, InvalidSessionException, UserNotFoundException, ProcessDefinitionNotFoundException {
-        return true;
-        //TODO: restore once BS-15125 has been fixed
-        //        final ProcessAPI processAPI = getBpmEngineAPIUtil().getProcessAPI(session);
-        //        boolean involvedInProcessInstance = processAPI.isInvolvedInProcessInstance(session.getUserId(), processInstanceID);
+        final ProcessAPI processAPI = getBpmEngineAPIUtil().getProcessAPI(session);
+        return processAPI.isInvolvedInProcessInstance(session.getUserId(), processInstanceID);
+        //restore this if the manager of a user should see his instances
```
Comment (laurentleseigneur) [inline forms/forms-server/src/main/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImpl.java]: shoul we keep this commented code about manager of user?

Comment (abirembaut) [inline forms/forms-server/src/main/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImpl.java]: there is the same commented code for the method canUserSeeTaskInstance.
I'm OK to remove it but if we do, it should be done for both methods

File: forms/forms-server/src/test/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImplTest.java [hunk 2]
```diff
@@ -77,16 +76,12 @@ public void setUp() throws Exception {
         doReturn(processApi).when(bpmEngineAPIUtil).getProcessAPI(session);
     }
 
-    //TODO restore once BS-15125 has been fixed
-    @Ignore
     @Test
     public void should_canUserSeeProcessInstance_call_engine_api() throws Exception {
         final boolean expected = true;
```
Comment (laurentleseigneur) [inline forms/forms-server/src/test/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImplTest.java]: can be inlined

File: forms/forms-server/src/test/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImplTest.java [hunk 3]
```diff
@@ -77,16 +76,12 @@ public void setUp() throws Exception {
...
         checkCanUserSeeProcessInstanceWhenApiReturn(expected);
     }
 
-    //TODO restore once BS-15125 has been fixed
-    @Ignore
     @Test
     public void should_canUserSeeProcessInstance_call_engine_api_false() throws Exception {
         final boolean expected = false;
```
Comment (laurentleseigneur) [inline forms/forms-server/src/test/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImplTest.java]: can be inlined

**Comment by laurentleseigneur** [comment]
ignored test are now passing using engine including engine-sp PR#228

File: forms/forms-server/src/main/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImpl.java (hunk 1 above)
Comment (laurentleseigneur) [inline forms/forms-server/src/main/java/org/bonitasoft/forms/server/api/impl/FormWorkflowAPIImpl.java]: this gave me the feeling that this code was expected by engine's method which is false

**Comment by abirembaut** [comment]
Changes due to feedback done

**Pull request merged**